dhall/tests/Dhall/Test/QuickCheck.hs

```diff
@@ -275,17 +285,31 @@ instance (Arbitrary s, Arbitrary a) => Arbitrary (Expr s a) where
             standardizedExpression
       where
         customGens
-            :: Gen Integer    -- Generates all Integer fields in Expr
-            :+ Gen Text.Text  -- Generates all Text fields in Expr
+            :: Gen Integer  -- Generates all Integer fields in Expr
+            :+ Gen Text     -- Generates all Text fields in Expr
+            :+ ConstrGen "Lam" 0 Text
+            :+ ConstrGen "Pi" 0 Text
+            :+ ConstrGen "Field" 1 Text
+            :+ ConstrGen "Project" 1 (Either (Set Text) (Expr s a))
             :+ ()
         customGens =
                integer
                -- 'Lam's and 'Pi's are encoded differently when the binding is
                -- the special string "_", so we generate some of these strings
                -- to improve test coverage for these code paths.
             :+ Test.QuickCheck.oneof [pure "_", arbitrary]
```

Comment:
It seems to me that this generator might override the `ConstrGen` generators below. At least I can still reproduce https://github.com/dhall-lang/dhall-haskell/issues/1941 when it's there, and removing it seems to fix #1941.

(CC @Lysxia)

Reply:
Yeah, the first match wins. generic-random's doc could be clearer on that point.

Reply:
Thanks, @Lysxia! :)

In that case it probably makes sense to move these field-type-specific generators below the `ConstrGen` generators.